src/otx/v2/adapters/torch/mmengine/mmaction/modules/models/backbones/movinet.py

```diff
@@ -0,0 +1,780 @@
+            cfg.expanded_channels,
+            cfg.out_channels,
+            kernel_size=(1, 1, 1),
+            padding=(0, 0, 0),
+            conv_type=conv_type,
+            tf_like=tf_like,
+            norm_layer=norm_layer,
+            activation_layer=nn.Identity,
+        )
+
+        if not (cfg.stride == (1, 1, 1) and cfg.input_channels == cfg.out_channels):
+            if cfg.stride != (1, 1, 1):
+                if tf_like:
+                    layers.append(TFAvgPool3D())
+                else:
+                    layers.append(nn.AvgPool3d((1, 3, 3), stride=cfg.stride, padding=cfg.padding_avg))
+            layers.append(
+                ConvBlock3D(
+                    in_planes=cfg.input_channels,
+                    out_planes=cfg.out_channels,
+                    kernel_size=(1, 1, 1),
+                    padding=(0, 0, 0),
+                    norm_layer=norm_layer,
+                    activation_layer=nn.Identity,
+                    conv_type=conv_type,
+                    tf_like=tf_like,
+                ),
+            )
+            self.res = nn.Sequential(*layers)
+        # ReZero
+        self.alpha = nn.Parameter(torch.tensor(0.0), requires_grad=True)
+
+    def forward(self, x: Tensor) -> Tensor:
+        """Forward function of BasicBneck."""
+        residual = self.res(x) if self.res is not None else x
+        if hasattr(self, "expand"):
+            x = self.expand(x)
+        x = self.deep(x)
+        x = self.se(x)
+        x = self.project(x)
+        return residual + self.alpha * x
+
+
+class MoViNet(nn.Module):
+    """MoViNet class used for video classification."""
+
+    def __init__(
+        self,
+        cfg: Config,
+        conv_type: str | None = "3d",
+        tf_like: bool | None = False,
+    ) -> None:
+        """Initialization.
+
+        Args:
+            cfg (Config): Configuration object containing network's hyperparameters.
+            conv_type (str, optional): A string indicating the type of convolutional layer
+                to use. Can be "2d" or "3d". Defaults to "3d".
+            tf_like (bool, optional): A boolean indicating whether to use TensorFlow like
+                convolution padding or not. Defaults to False.
+
+        Attributes:
+            conv1 (ConvBlock3D): A convolutional block for the first layer.
+            blocks (nn.Sequential): A sequence of basic bottleneck blocks.
+            conv7 (ConvBlock3D): A convolutional block for the final layer.
+
+        Methods:
+            avg(x: Tensor) -> Tensor: A static method that returns the adaptive average pool
+                of the input tensor.
+            _init_weights(module): A private method that initializes the weights of the network's
+                convolutional, batch normalization, and linear layers.
+            forward(x: Tensor) -> Tensor: The forward pass of the network.
+        """
+        super().__init__()
+        tf_like = True
+        blocks_dic = OrderedDict()
```

Comment:
it would be easier to follow
```suggestion
        blocks_dict = OrderedDict()
```

Reply:
Done. https://github.com/openvinotoolkit/training_extensions/pull/2627/commits/7169459c0e1c71d2ee480b906c4533af36004ca9